### PR TITLE
add $resolved property to get quick access to $loaded state

### PR DESCRIPTION
**Pull request by soumak77**
addressed https://github.com/firebase/angularfire/issues/836

### CODE REVIEW

**Comment by soumak77** [comment]
I signed it!

**Comment by katowulf** [comment]
This is redundant. The functionality being added here is obtained by simply calling $loaded(), which is essentially a getter method of the promise you're looking to add.

**Comment by soumak77** [comment]
@katowulf that is not the case, please see my comment on #836

**Comment by jwngr** [comment]
Looks like there is a merge conflict. Can you please merge this with `master`?

**Comment by soumak77** [comment]
@jwngr done

**Comment by katowulf** [comment]
Failed tests looks like some weird SSH issue. Retrying. If they fail again I'll post more here for review.

**Comment by katowulf** [comment]
@jwngr any ideas on this one?

> The command "openssl aes-256-cbc -K $encrypted_d1b4272f4052_key -iv $encrypted_d1b4272f4052_iv -in tests/key.json.enc -out tests/key.json -d" failed and exited with 1 during .

File: src/database/FirebaseArray.js [hunk 1]
```diff
@@ -88,6 +88,13 @@
 
         this._sync.init(this.$list);
 
+        // $resolved provides quick access to the current state of the $loaded() promise.
+        // This information is useful in data-binding for knowing when the data is available.
```
Comment (katowulf) [inline src/database/FirebaseArray.js]: nitpick: this line is superfluous and adds no value (is actually a bit confusing)

Comment (katowulf) [inline src/database/FirebaseArray.js]: (An example usage would improve on this.)

**Comment by katowulf** [comment]
lgtm

**Comment by jwngr** [comment]
This is because we currently require a service account to run the tests and the CI tests only work if a member with commit access kicks of the request. That being said, I've meant to remove this restriction and will do so tonight.

Assigning this to myself so I can remember to do this.

**Comment by soumak77** [comment]
@katowulf I updated the code with an example.  The example uses both JS and HTML, so I'm not sure if it follows the desired style guide.

**Comment by jwngr** [comment]
Thanks, but I think we'd prefer it if we don't include an example like that in the code itself. It takes up a ton of space and that info belongs in the docs anyway. I plan to merge this is as you had it (so please remove the latest commit) and I will add the method to our docs reference page.

**Comment by jwngr** [comment]
This looks good to me as well so I'm going to merge this in. Thanks for your contribution @soumak77!

Note that I just put together #850 which will fix the issue with the failing CI. But I just tested this change locally and all the tests pass, so I'm fine merging it in despite the failing CI.

**Comment by soumak77** [comment]
@jwngr any chance you could push out a patch release with this change?

**Comment by jwngr** [comment]
Yup, sorry about that. I just put together https://github.com/firebase/angularfire/pull/874 which documents this feature and adds a changelog for a `2.1.0` release. Once that gets merged in, I'll cut a new release. Hopefully should be done by the end of the day.

**Comment by jwngr** [comment]
@soumak77 - [AngularFire `2.1.0`](https://github.com/firebase/angularfire/releases/tag/v2.1.0) is out with your change! Thanks again.

**Comment by soumak77** [comment]
Awesome, thanks @jwngr!

**Pull request merged**